Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1162,9 +1172,94 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
     return false;
 }
 
+#pragma mark Selections
+
+auto UnifiedPDFPlugin::selectionGranularityForMouseEvent(const WebMouseEvent& event) const -> SelectionGranularity
+{
+    if (event.clickCount() == 2)
+        return SelectionGranularity::Word;
+    if (event.clickCount() == 3)
+        return SelectionGranularity::Line;
+    return SelectionGranularity::Character;
+}
+
+void UnifiedPDFPlugin::beginTrackingSelection(PDFDocumentLayout::PageIndex pageIndex, const WebCore::IntPoint& pagePoint, SelectionGranularity selectionGranularity, OptionSet<WebEventModifier> mouseEventModifiers)
+{
+    m_selectionTrackingData.isActive = true;
+    m_selectionTrackingData.granularity = selectionGranularity;
+    m_selectionTrackingData.startPageIndex = pageIndex;
+    m_selectionTrackingData.startPagePoint = pagePoint;
+    m_selectionTrackingData.marqueeSelectionRect = { };
+    m_selectionTrackingData.shouldMakeMarqueeSelection = mouseEventModifiers.contains(WebEventModifier::AltKey);
+    m_selectionTrackingData.shouldExtendCurrentSelection = mouseEventModifiers.contains(WebEventModifier::ShiftKey);
+
+    if (m_selectionTrackingData.shouldMakeMarqueeSelection)
+        return continueTrackingSelection(pageIndex, pagePoint);
+
+    // FIXME: <https://webkit.org/b/268617>  Selections should be extensible on shift-click.
+    if (m_selectionTrackingData.shouldExtendCurrentSelection)
+        notImplemented();
+
+    continueTrackingSelection(pageIndex, pagePoint);
```

Comment:
Will rewrite this as, since `continueTrackingSelection` does the same check, and there's no reason to bail early here -- we're not doing anything right now anyway.
```suggestion
    // FIXME: <https://webkit.org/b/268617>  Selections should be extensible on shift-click.
    if (m_selectionTrackingData.shouldExtendCurrentSelection)
        notImplemented();

    continueTrackingSelection(pageIndex, pagePoint);
```

Reply:
Fixed in [53cae5f](https://github.com/WebKit/WebKit/commit/53cae5fe11eca85d9ea822cdec0181e4a4f4dcf6)